pylint/pyreverse/inspector.py

```diff
@@ -6,22 +6,30 @@
 
 Try to resolve definitions (namespace) dictionary, relationship...
 """
+from __future__ import annotations
```

Comment:
```suggestion

from __future__ import annotations
```

Reply:
Some day I will learn this! 😄 
Maybe that could be a pre-commit check or pylint extension as well?

Reply:
I'm curious, what is the source for this convention?

Reply:
PEP8 seems to say the opposite (emphasis mine): Imports are always put at the top of the file, _just after_ any module comments and docstrings, and before module globals and constants.

Reply:
It's my interpretation of PEP8's: `You should put a blank line between each group of imports.` Since the `module docstring` is certainly not the same "group" as the future import it makes sense the separate them just like you separate the last import from the first definition. 

Note the example just below your quote:
```python
"""This is the example module.

This module does stuff.
"""

from __future__ import barry_as_FLUFL

__all__ = ['a', 'b', 'c']
__version__ = '0.1'
__author__ = 'Cardinal Biggles'

import os
import sys
```

https://peps.python.org/pep-0008/ for reference.


But, I agree this is somewhat arbitrary. There does seem to be consensus though as https://github.com/psf/black/pull/2996 seems to have general support and would help us with this as well.

Reply:
Ah, yeah I wondered if that's what you were picking up on. I don't think the document is very clear. I think it's not clear that it's incompatible with "put a blank line before each subsequent group of imports". I'm not going to open a PR against PEP8 (😅 ) but maybe I'll chime in on the black issue, thanks for posting it!

Reply:
I do think a suggestion to edit the PEP and linking to `black` wouldn't hurt. Just to see if there is actual agreement on it.

If `black` is going to enforce it it will be pretty "standard" anyway.

Reply:
I guess I'm feeling we shouldn't enforce this style in pylint or do comments on it. If `black` changes, we'll get it for free. That example is showing a blank line at the top, but it's not "between" two groups of imports, it's between a docstring, so it's just a random blank line. But no big deal either way obviously.

Reply:
Fine with me! Agree that increasing maintenance churn for something that is likely auto-fixed by `black` in a couple of weeks is not worth the effort!

Reply:
isort is the auto-formatter that would enforce this I think.